tests/functions.sh

```diff
@@ -64,14 +71,48 @@ function ssh_no_ping {
     fi
 }
 
+function ssh_udp {
+    id1=$1
+    id2=$2
+    port=$3
+    ssh_run_background $id1 "nc -w 1 -lup $port > /tmp/test"
+    ssh_run_background $id2 "echo 'this message is from vm $id2' | nc -u -w 1 42.0.0.$id1 $port"
+    ssh_run $id2 "sleep 1"
+    ssh_run $id1 [ -s "/tmp/test" ]
+    if [ "$?" == "0" ]; then
+	echo "udp test $id2 --> $id1 OK"
+    else
+	echo "udp test $id2 --> $id1 FAIL"
+	RETURN_CODE=1
+    fi
+    ssh_run $id1 "rm /tmp/test"
+}
+
+function ssh_no_udp {
+    id1=$1
+    id2=$2
+    port=$3
+    ssh_run_background $id1 "nc -w 1 -lup $port > /tmp/test"
+    ssh_run_background $id2 "echo 'this message is from vm $id2' | nc -u -w 1 42.0.0.$id1 $port"
+    ssh_run $id2 "sleep 1"
+    ssh_run $id1 [ -s "/tmp/test" ]
```

Comment:
This code is exactly the same that the one above, I guess you will have a similar code in ssh_tcp too, so you can create a function which start the nc server and client, and return true if the server have receive the message.